packages/trader/src/Modules/Trading/Components/Form/WheelPicker/wheel-picker-mobile.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useSwipeable } from 'react-swipeable';
+import { Text } from '@deriv/components';
+import { motion } from 'framer-motion';
+
+type WheelPickerMobileProps = {
+    defaultValue?: number;
+    onChange?: (val: number) => void;
+    options: number[];
+    currency: string;
+    optionHeight?: number;
+};
+
+type SwipeState = {
+    startY: number;
+    deltaY: number;
+    translateY: number;
+};
+
+export function getTargetIndex({
+    deltaY,
+    snapTolerance,
+    optionHeight,
+    options,
+    selectedIndex,
+}: {
+    deltaY: number;
+    snapTolerance: number;
+    optionHeight: number;
+    options: number[];
+    selectedIndex: number;
+}) {
+    const absDeltaY = Math.abs(deltaY);
+    const yLeftover = absDeltaY % optionHeight;
+    const snapThreshold = Math.round(optionHeight * snapTolerance);
+    const snapY = Number(yLeftover >= snapThreshold);
+
+    const absDeltaIndex = Math.floor(absDeltaY / optionHeight) + snapY;
+    const deltaIndex = absDeltaIndex * Math.sign(deltaY) * -1;
+
+    const MIN_INDEX = 0;
+    const MAX_INDEX = options.length - 1;
+    const targetIndex = Math.min(MAX_INDEX, Math.max(MIN_INDEX, selectedIndex + deltaIndex));
+    return targetIndex;
+}
+
+const WheelPickerMobile: React.FC<WheelPickerMobileProps> = ({
+    defaultValue,
+    onChange,
+    currency,
+    options,
+    optionHeight = 28,
+}) => {
+    const [selectedIndex, setSelectedIndex] = useState<number>(options.length - 1);
+    const [optionWidth, setOptionWidth] = useState<number>(0);
+
+    const [swipe, setSwipe] = useState<SwipeState>({ startY: 0, deltaY: 0, translateY: 0 });
+    const optionRef = useRef<HTMLDivElement>(null);
+
+    const swipeableHandlers = useSwipeable({
+        onSwipeStart: () => {
+            setSwipe(swipe => ({ ...swipe, startY: swipe.translateY }));
+        },
+        onSwiping: ({ deltaY, first }) => {
+            if (first) return;
+            const MIN = 0;
+            const MAX = optionHeight * (options.length - 1);
+            const translateY = Math.min(MAX, Math.max(MIN, swipe.startY + deltaY));
+            const newIndex = Math.min(options.length - 1, Math.max(0, Math.round(translateY / optionHeight)));
+            setSwipe(swipe => ({
+                ...swipe,
+                deltaY,
+                translateY: (translateY + optionHeight * newIndex) / 2,
+            }));
+            setSelectedIndex(newIndex);
+        },
+        onSwiped: ({ deltaY }) => {
+            const MIN = 0;
+            const MAX = optionHeight * (options.length - 1);
```

Comment:
@ahmadtaimoor-deriv i think we're declaring MIN & MAX twice but they depend on the props only, so we can declare them once in the upper scope, right? :)

Reply:
A generic function added , thanks